tools/build.py

```diff
@@ -0,0 +1,681 @@
+        for config in configs:
+            os.makedirs(_get_build_config_dir(build_dir, config), exist_ok=True)
+
+    log.info("Build started")
+
+    if args.update:
+        if _is_reduced_ops_build(args):
+            log.info("Generating config for selected ops")
+            _generate_selected_ops_config(args.include_ops_by_config)
+
+        cmake_extra_args = []
+
+        if is_windows():
+            cpu_arch = platform.architecture()[0]
+            if args.build_wasm:
+                cmake_extra_args = ["-G", "Ninja"]
+            elif args.cmake_generator == "Ninja":
+                if cpu_arch == "32bit" or args.arm or args.arm64 or args.arm64ec:
+                    raise UsageError(
+                        "To cross-compile with Ninja, load the toolset environment for the target processor "
+                        "(e.g. Cross Tools Command Prompt for VS)")
+                cmake_extra_args = ["-G", args.cmake_generator]
+            elif args.arm or args.arm64 or args.arm64ec:
+                # Cross-compiling for ARM(64) architecture
+                if args.arm:
+                    cmake_extra_args = ["-A", "ARM"]
+                elif args.arm64:
+                    cmake_extra_args = ["-A", "ARM64"]
+                elif args.arm64ec:
+                    cmake_extra_args = ["-A", "ARM64EC"]
+                cmake_extra_args += ["-G", args.cmake_generator]
+
+                # Cannot test on host build machine for cross-compiled
+                # builds (Override any user-defined behaviour for test if any)
+                if args.test:
+                    log.warning("Cannot test on host build machine for cross-compiled ARM(64) builds. "
+                                "Will skip test running after build.")
+                    args.test = False
+            elif cpu_arch == "32bit" or args.x86:
+                cmake_extra_args = ["-A", "Win32", "-T", "host=x64", "-G", args.cmake_generator]
+            else:
+                toolset = "host=x64"
+
+                # TODO: Do we need the ability to specify the toolset?
+                # if args.msvc_toolset:
+                #     toolset += f",version={args.msvc_toolset}"
+
+                cmake_extra_args = ["-A", "x64", "-T", toolset, "-G", args.cmake_generator]
+        elif args.cmake_generator is not None and not (is_macOS() and args.use_xcode):
+            cmake_extra_args += ["-G", args.cmake_generator]
+        elif is_macOS():
+            if args.use_xcode:
+                cmake_extra_args += ["-G", "Xcode"]
+            if not args.ios and not args.android and args.osx_arch == "arm64" and platform.machine() == "x86_64":
+                if args.test:
+                    log.warning("Cannot test ARM64 build on X86_64. Will skip test running after build.")
+                    args.test = False
+
+        if args.build_wasm:
+            emsdk_version = args.emsdk_version
+            emsdk_dir = REPO_DIR / "cmake" / "external" / "emsdk"
```

Comment:
this path doesn't exist. do we need `--build_wasm`?

Reply:
@guschmue are you folks likely to want to use build.py in a development enivronment to build ort-extensions? If not I will remove the wasm related stuff from here.

Reply:
Simplified the wasm setup to be the minimal required for now. emsdk needs to be manually cloned.